src/XrmContext/Resources/XrmExtensions.cs

```diff
@@ -400,4 +400,24 @@ public static ColumnSet GetColumnSet<T>(params Expression<Func<T, object>>[] att
             return attrGetters.Select(a => GetAttributeLogicalName(a).ToLower()).Any(a => entity.Attributes.Remove(a));
         }
     }
+
+    [AttributeUsage(AttributeTargets.Field)]
+    public class OptionSetMetadataAttribute : Attribute
```

Comment:
`ExternalValue` could probably be removed from this class since XrmContext never generates this anyway 🤷‍♂️